File: leetcode/1-two-sum/two-sum.php
class Solution {

    /**
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer[]
     */
    function twoSum($nums, $target) {
        $map = [];

        foreach ($nums as $i => $num) {
            $complement = $target - $num;
            if (isset($map[$complement])) {
                return [$map[$complement], $i];
            }
            $map[$num] = $i;
        }

        return []; 
    }
}

File: leetcode/1-two-sum/two-sum.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var seen = [Int: Int]()  // Dictionary to store number and its index

        for (i, num) in nums.enumerated() {
            let complement = target - num
            if let j = seen[complement] {
                return [j, i]
            }
            seen[num] = i
        }

        return [] 
    }
}
